velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -394,6 +395,20 @@ parquet::WriterOptions getParquetOptions(
     parquetOptions.parquetWriteTimestampUnit =
         options.parquetWriteTimestampUnit.value();
   }
+  if (options.parquetVersionString.has_value()) {
+
+    if (options.parquetVersionString.value() == "PARQUET_1_0")
```

Comment:
We might not actually pass around a string with the value from the config. When the session comes in we could map that to a numeric constant. 
Also this can be simplified to
```
if (options.parquetVersionString.value() == "PARQUET_2_6")
{
   parquetOptions.parquetVersion = arrow::ParquetVersion::PARQUET_2_6;
}
```
with a comment that the default is arrow::ParquetVersion::PARQUET_1_0.

Reply:
Yeah, the conditional was needlessly bulky, cleaned that up and added comment. 

I passed around the string because the handy `arrow::ParquetVersion::type` enum is inaccessible from `HiveDataSink.cpp`, which reads session properties and catalog config to prepare the writer. Though it seems that the only permitted version number that can really be passed in via session/config is `1.0`, due to the discrepancy between Java's Apache and C++ Arrow. 

If we don't plan to change that, then this `options.parquetVersionString` can be turned into a simple boolean, to toggle version `1.0` if necessary, or leave it at the default `2.6` otherwise.

Reply:
Perhaps someone else has a comment but bool doesn't seem a good idea as this can change in the future.
How about defining an enum in the common::WriterOptions that stores the consts to represent parquet versions supported by the writer? In that way they are localized to where they are used but also allows for expansion or refactoring should it become necessary.